### PR TITLE
Support Cockroach db

### PR DESCRIPTION
**Pull request by jiqiang90**
# Description
Support cockroach db

await merge of https://github.com/subquery/graphile-engine/pull/3/files#diff-625984a1091f07471235c0bdbe8593f1b41f7712ce92bb5bd2abfa921991d445

And update dependencies

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] I have tested locally
- [ ] I have performed a self review of my changes
- [ ] Updated any relevant documentation
- [ ] Linked to any relevant issues
- [ ] I have added tests relevant to my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] My code is up to date with the base branch

### CODE REVIEW

File: packages/node-core/src/db/db.module.ts [hunk 1]
```diff
@@ -25,24 +25,33 @@ const DEFAULT_DB_OPTION: DbOption = {
...
+async function establishConnectionSequelize(option: SequelizeOption, numRetries: number): Promise<Sequelize> {
+  const uri = `postgresql://${option.username}:${option.password}@${option.host}:${option.port}/${option.database}`;
+
+  const sequelize = new Sequelize(uri, option);
   try {
     await sequelize.authenticate();
   } catch (error) {
-    logger.error(error, 'Unable to connect to the database');
+    if (JSON.stringify(error.message).includes(CONNECTION_SSL_ERROR_REGEX)) {
+      logger.warn('Database not support SSL connection, will try to connect without it');
```
Comment (stwiname) [inline packages/node-core/src/db/db.module.ts:38]:
```suggestion
      logger.warn('Database does not support SSL connection, will try to connect without it');
```

File: packages/node-core/src/configure/NodeConfig.ts [hunk 2]
```diff
@@ -63,6 +63,7 @@ const DEFAULT_CONFIG = {
   disableHistorical: false,
   multiChain: false,
   unfinalizedBlocks: false,
+  enableSSL: false,
```
Comment (stwiname) [inline packages/node-core/src/configure/NodeConfig.ts:66]: Does this conflict with https://github.com/subquery/subql/pull/1513

File: packages/node-core/src/indexer/store.service.ts [hunk 3]
```diff
@@ -110,6 +112,23 @@ export class StoreService {
...
+      logger.warn(`Subscription is not support with ${this.dbType}`);
+    }
+  }
+
+  async getDbType(): Promise<SUPPORT_DB> {
+    if (!this.dbType) {
+      const [result] = await this.sequelize.query('select version()');
+      const matchDB = Object.values(SUPPORT_DB).find((db) => (result[0] as {version: string}).version.includes(db));
+      if (!matchDB) {
+        throw new Error(`Database not supported, got ${result}`);
```
Comment (stwiname) [inline packages/node-core/src/indexer/store.service.ts:127]:
```suggestion
        throw new Error(`Database type not supported, got ${result}`);
```

File: packages/node-core/src/indexer/store.service.ts [hunk 4]
```diff
@@ -120,6 +139,11 @@ export class StoreService {
   }
 
   async initHotSchemaReloadQueries(schema: string): Promise<void> {
+    if (this.dbType === SUPPORT_DB.cockRoach) {
+      logger.warn(`Hot schema reload feature is not support with ${this.dbType}`);
```
Comment (stwiname) [inline packages/node-core/src/indexer/store.service.ts:143]:
```suggestion
      logger.warn(`Hot schema reload feature is not supported with ${this.dbType}`);
```

File: packages/query/src/graphql/graphql.module.ts [hunk 5]
```diff
@@ -55,6 +55,11 @@ export class GraphqlModule implements OnModuleInit, OnModuleDestroy {
     } catch (e) {
       throw new Error(`create apollo server failed, ${e.message}`);
     }
+    if ((await this.projectService.getDbType()) === SUPPORT_DB.cockRoach) {
+      logger.info(`Using Cockroach database, subscription and host-schema functions are not supported`);
```
Comment (stwiname) [inline packages/query/src/graphql/graphql.module.ts:59]:
```suggestion
      logger.info(`Using Cockroach database, subscription and hot-schema functions are not supported`);
```

File: packages/query/src/graphql/project.service.ts [hunk 6]
```diff
@@ -2,12 +2,26 @@
...
 export class ProjectService {
   constructor(private readonly pool: Pool, private readonly config: Config) {}
+  private dbType: SUPPORT_DB;
+
+  async getDbType(): Promise<SUPPORT_DB> {
+    if (!this.dbType) {
+      const result = await this.pool.query<{version: string}>('select version()');
+      const matchDB = Object.values(SUPPORT_DB).find((db) => result.rows[0].version.includes(db));
+      if (!matchDB) {
+        throw new Error(`Database not supported, got ${result}`);
```
Comment (stwiname) [inline packages/query/src/graphql/project.service.ts:19]:
```suggestion
        throw new Error(`Database type not supported, got ${result}`);
```

File: packages/query/src/graphql/project.service.ts [hunk 7]
```diff
@@ -2,12 +2,26 @@
...
+import {SUPPORT_DB} from '@subql/common';
 import {Pool} from 'pg';
 import {Config} from '../configure';
 
 @Injectable()
 export class ProjectService {
   constructor(private readonly pool: Pool, private readonly config: Config) {}
+  private dbType: SUPPORT_DB;
+
+  async getDbType(): Promise<SUPPORT_DB> {
```
Comment (stwiname) [inline packages/query/src/graphql/project.service.ts:14]: Could we move this into common so its not duplicated on node and query?

File: packages/node-core/src/configure/NodeConfig.ts (hunk 2 above)
Comment (jiqiang90) [reply packages/node-core/src/configure/NodeConfig.ts:66]: This actually not been used, remove it now.

I checked the PR, looks like not conflicting, now we are enable SSL connection by default, if DB not support his feature will try connect again without SSL. I think another PR can make change base on this one.

File: packages/node-core/src/db/db.module.ts [hunk 8]
```diff
@@ -57,6 +66,7 @@ export class DbModule {
       ssl: nodeConfig.isPostgresSecureConnection,
       dialectOptions: {
         ssl: {
+          rejectUnauthorized: false,
```
Comment (stwiname) [inline packages/node-core/src/db/db.module.ts:69]: What happens now if we connect with ssl will it fail?

**Comment by jiqiang90** [comment]
Currently comment on enum is not supported by cockroach db https://github.com/cockroachdb/cockroach/issues/44135. 

This affect the name display for enums on query service. Once this supported, we can enable it

**Review by stwiname — APPROVED** [review]

**Pull request merged**